tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordedTestSanitizer.cs

```diff
@@ -110,17 +113,20 @@ public virtual void SanitizeBody(RequestOrResponse message)
 
         public virtual void Sanitize(RecordEntry entry)
         {
-            entry.RequestUri = SanitizeUri(entry.RequestUri);
+            if (Condition.Applicable(entry))
```

Comment:
If we update so that Condition can be null, this could change to:
if (Condition?.Applicable(entry) == true)

Reply:
You're right. `if (Condition?.Applicable(entry) == true || Condition == null)`

👍

Reply:
You shouldn't need the Condition == null part because if it is null, Condition?.Applicable will not evaluate to true.

Reply:
oh.. my bad - if it is null that means we _do_ need to go into the if block.

Reply:
In that case, you can just swap the order and avoid the null conditional:
```c#
if (Condition == null || Condition.Applicable(entry))
```